hudi-common/src/main/java/org/apache/hudi/common/table/read/UpdateProcessor.java

```diff
@@ -95,15 +102,15 @@ public CallbackProcessor(BaseFileUpdateCallback callback, UpdateProcessor<T> del
     }
 
     @Override
-    public T processUpdate(String recordKey, T previousRecord, T currentRecord, boolean isDelete) {
-      T result = delegate.processUpdate(recordKey, previousRecord, currentRecord, isDelete);
+    public BufferedRecord<T> processUpdate(String recordKey, BufferedRecord<T> previousRecord, BufferedRecord<T> currentRecord, boolean isDelete) {
+      BufferedRecord<T> result = delegate.processUpdate(recordKey, previousRecord, currentRecord, isDelete);
 
       if (isDelete) {
-        callback.onDelete(recordKey, previousRecord);
-      } else if (previousRecord != null && previousRecord != currentRecord) {
-        callback.onUpdate(recordKey, previousRecord, currentRecord);
+        callback.onDelete(recordKey, previousRecord.getRecord());
+      } else if (previousRecord != null && previousRecord.getRecord() != currentRecord.getRecord()) {
```

Comment:
Can we update the BaseFileUpdateCallback to also operate on the BufferedRecord so we can use the operation type?

Reply:
Seems BaseFileUpdateCallback does not need operation type in BufferedRecord ?